### ISSUE TITLE
[examples/text-classification] `do_predict` for the test set of local datasets

### ISSUE THREAD

**Issue by forest1988** [huggingface/transformers#9442]
# 🚀 Feature request

It seems that `run_glue.py` has the train set and the validation set management for local CSV/JSON files, but it doesn't have args for managing the test set of the local datasets.

https://github.com/huggingface/transformers/blob/7a9f1b5c99e9a5d1772649d029acdf5160419239/examples/text-classification/run_glue.py#L90-L95

I think the script is intended to be used not only for the train/validation but the test, as `glue` tasks test sets are downloaded as shown in https://huggingface.co/docs/datasets/loading_datasets.html#selecting-a-configuration.
It has the `--do_predict` option for the test sets.

If there is no particular reason for not having the ability to read the test set in the local dataset, would it be ok for me to add the feature?
Or is there some intention behind this implementation?

## Motivation

I'd like to train, validate, and test my own local dataset.

## Your contribution

I think some modifications like the below may help to add the feature.

``` python
    test_file: Optional[str] = field(
        default=None, metadata={"help": "A csv or a json file containing the test data."}
    )
```

``` python
        datasets = load_dataset(
            "csv", data_files={"train": data_args.train_file, "validation": data_args.validation_file, "test": data_args.test_file}
        )
```

``` python
    # if data_args.task_name is not None:
    #     test_dataset = datasets["test_matched" if data_args.task_name == "mnli" else "test"]
    test_dataset = datasets["test_matched" if data_args.task_name == "mnli" else "test"]
```

Thank you in advance.

**Comment by sgugger** [comment]
As long as it's kept simple (we want short and focused example scripts, so that they are easy to understand and tweak), I don't mind adding this feature. Feel free to open a PR with your suggestions!

**Comment by forest1988** [comment]
Thanks, I'll try a modification as simple as possible, and if it can fix this issue without making the example difficult to understand, I’ll open a PR!
